### PR TITLE
[CD] Enable pypi dependencies both for XPU linux and Windows whls

### LINKED ISSUE #135867: `import torch` fails on stock ubuntu if one uses XPU nightly binaries
### 🐛 Describe the bug

```bash
$ docker run --rm -it python:3.10 bash -c "pip install --pre torch --index-url https://download.pytorch.org/whl/nightly/xpu/; python -c 'import torch;print(torch.__version__)'"
Looking in indexes: https://download.pytorch.org/whl/nightly/xpu/
Collecting torch
  Downloading https://download.pytorch.org/whl/nightly/xpu/torch-2.5.0.dev20240912%2Bxpu-cp310-cp310-linux_x86_64.whl (803.0 MB)
...
Successfully installed MarkupSafe-2.1.5 filelock-3.13.1 fsspec-2024.6.1 jinja2-3.1.4 mpmath-1.3.0 networkx-3.3 packaging-22.0 pytorch-triton-xpu-3.1.0+91b14bf559 sympy-1.13.1 torch-2.5.0.dev20240912+xpu typing-extensions-4.12.2
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.10/site-packages/torch/__init__.py", line 367, in <module>
    _load_global_deps()
  File "/usr/local/lib/python3.10/site-packages/torch/__init__.py", line 323, in _load_global_deps
    raise err
  File "/usr/local/lib/python3.10/site-packages/torch/__init__.py", line 300, in _load_global_deps
    ctypes.CDLL(global_deps_lib_path, mode=ctypes.RTLD_GLOBAL)
  File "/usr/local/lib/python3.10/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libmkl_intel_lp64.so.2: cannot open shared object file: No such file or directory
```

### Versions

nightly

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @gujinghui @EikanWang @fengyuan14 @guangyey

### PR DESCRIPTION
**Pull request by chuanqi129**
Enable xpu runtime pypi packages as dependencies of XPU CD wheels both for Linux and Windows.
Fixes https://github.com/pytorch/pytorch/issues/135867
Works for https://github.com/pytorch/pytorch/issues/139722 and https://github.com/pytorch/pytorch/issues/114850

### CODE REVIEW

File: .github/scripts/generate_binary_build_matrix.py [hunk 1]
```diff
@@ -84,6 +84,19 @@
         "nvidia-nvtx-cu12==12.6.77; platform_system == 'Linux' and platform_machine == 'x86_64' | "
         "nvidia-nvjitlink-cu12==12.6.77; platform_system == 'Linux' and platform_machine == 'x86_64'"
     ),
+    "xpu": (
```
Comment (atalman) [inline .github/scripts/generate_binary_build_matrix.py:87]: Hi @chuanqi129 please add these packages here: https://github.com/pytorch/test-infra/edit/main/s3_management/manage.py

Comment (atalman) [reply .github/scripts/generate_binary_build_matrix.py:87]: And perhaps here as well: https://github.com/pytorch/test-infra/blob/main/s3_management/update_dependencies.py#L11 
So we can update them via script.

Comment (chuanqi129) [reply .github/scripts/generate_binary_build_matrix.py:87]: Thanks @atalman help to point it out, the PR has been submitted https://github.com/pytorch/test-infra/pull/5965. Please help to review it

**Comment by chuanqi129** [comment]
Hi @atalman, I have retriggered the CI test and all PR change related tests are passed, could you please help to review it again?

**Review by atalman — APPROVED** [review]
lgtm

**Pull request closed without merging**